src/demo/OceanObject.js

```diff
@@ -58,13 +58,31 @@ export const loadAllFishPartImages = () => {
   );
 };
 
-export const generateOceanObject = (allowedClasses, id) => {
+export const generateOceanObject = (allowedClasses, id, dataSet = null) => {
   const idx = Math.floor(Math.random() * allowedClasses.length);
-  const newOceanObject = new allowedClasses[idx](id);
+  const newOceanObject = new allowedClasses[idx](
+    id,
+    filterFishComponents(dataSet)
+  );
   newOceanObject.randomize();
   return newOceanObject;
 };
 
+const filterFishComponents = dataSet => {
+  if (!dataSet) {
+    return fishData;
+  }
+
+  let filteredCopy = {...fishData};
+  Object.keys(filteredCopy).forEach(key => {
+    filteredCopy[key] = Object.values(filteredCopy[key]).filter(
+      option => !(option.exclusions || []).includes(dataSet)
+    );
+  });
```

Comment:
It would be very good to unit test this logic now.

Reply:
good call, adding tests now

Reply:
Thanks for that.  I don't usually write a lot of tests, and when I do, they're usually to make sure things keep working after we're past a hectic creative phase, but whenever there is a tricky algorithm, I think it's great to use tests to help get it right up front, and to make sure it never regresses.  My 2c :)